x-pack/plugins/threat_intelligence/public/modules/cases/utils/attachments.tsx

```diff
@@ -0,0 +1,124 @@
+ * - the component that renders the comment in teh case attachment
+ */
+export const generateAttachmentType = (): ExternalReferenceAttachmentType => ({
+  id: ExternalAttachmentTypeId,
+  displayName: 'indicator',
+  getAttachmentViewObject: () => ({
+    event: (
+      <FormattedMessage
+        id="xpack.threatIntelligence.cases.eventDescription"
+        defaultMessage="added an indicator of compromise"
+      />
+    ),
+    timelineAvatar: <EuiAvatar name="indicator" color="subdued" iconType="crosshairs" />,
+    children: React.lazy(async () => {
+      const { initComponent } = await import(
+        '../components/attachment_children/attachment_children'
+      );
+      return {
+        default: initComponent(),
+      };
+    }),
+  }),
+  icon: 'crosshairs',
+});
+
+/**
+ * Creates an attachment object, then passed to a case. It contains all the information necessary (id and metadata)
+ * to allow the lazy loaded component defined within {@link generateAttachmentType} to render.
+ *
+ * See documentation here https://docs.elastic.dev/reops/attachment-framework-external-reference
+ *
+ * @param externalReferenceId the id saved in the case's attachment (in our case the indicator id)
+ * @param attachmentMetadata some metadata also saved in the case's attachments for display in the comment
+ */
+export const generateAttachmentsWithoutOwner = (
+  externalReferenceId: string,
+  attachmentMetadata: AttachmentMetadata
+): CaseAttachmentsWithoutOwner => {
+  if (!externalReferenceId) {
+    return [];
+  }
+
+  return [
+    {
+      type: CommentType.externalReference,
+      externalReferenceId,
+      externalReferenceStorage: {
+        type: ExternalReferenceStorageType.elasticSearchDoc,
+      },
+      externalReferenceAttachmentTypeId: ExternalAttachmentTypeId,
+      externalReferenceMetadata: attachmentMetadata as unknown as { [p: string]: JsonValue },
+    },
+  ];
+};
+
+/**
+ * To facilitate the rendering of the lazy loaded component defined in {@link generateAttachmentType}, we pass
+ * some of the indicator's fields to be saved in the case's attachment.
+ *
+ * @param indicator the indicator we're attaching to a case
+ */
+export const generateAttachmentsMetadata = (indicator: Indicator): AttachmentMetadata => {
+  const indicatorName: string | null = getIndicatorFieldAndValue(
+    indicator,
+    RawIndicatorFieldId.Name
+  ).value;
+  const indicatorType: string | null = getIndicatorFieldAndValue(
+    indicator,
+    RawIndicatorFieldId.Type
+  ).value;
+  const indicatorFeedName: string | null = getIndicatorFieldAndValue(
+    indicator,
+    RawIndicatorFieldId.Feed
+  ).value;
+  const indicatorFirstSeen: string | null = getIndicatorFieldAndValue(
+    indicator,
+    RawIndicatorFieldId.FirstSeen
+  ).value;
+
+  return {
+    indicatorName: indicatorName || '',
+    indicatorType: indicatorType || '',
+    indicatorFeedName: indicatorFeedName || '',
+    indicatorFirstSeen: indicatorFirstSeen || '',
```

Comment:
we do in the flyout, and because the rawDocument we're getting back when fetching by id is different, it was a lot easier to use this. I explained it in the code documentation of that interface but I realize it's not super clear (but I didn't want to be too precise if we decided to add/remove fields in the future)